r/NEWS.md

```diff
@@ -59,6 +59,7 @@ See `vignette("install", package = "arrow")` for details.
 * Factor types are now preserved in round trip through Parquet format (#6135, @yutannihilation)
 * Reading an Arrow dictionary type coerces dictionary values to `character` (as R `factor` levels are required to be) instead of raising an error
 * Many improvements to Parquet function documentation (@karldw, @khughitt)
+* Can now infer the type of an R `list` and create a ListArray when all list elements are the same type (#6275, @michaelchirico)
```

Comment:
@fsaintjacques could you please move this bullet up to the current release (L23) (since this did not get included in 0.16.0)?

Other than that, if you're done here, please merge. Merci!